e2e/main_test.go

```diff
@@ -65,7 +69,13 @@ func TestMain(m *testing.M) {
 }
 
 func createNamespace(c kubernetes.Interface, name string) (*v1.Namespace, error) {
-	namespace, err := c.CoreV1().Namespaces().Create(&v1.Namespace{
+	namespace, err := c.CoreV1().Namespaces().Get(name, metav1.GetOptions{})
```

Comment:
Is this strictly necessary? You can also just create and check if the error is https://godoc.org/k8s.io/apimachinery/pkg/api/errors#IsAlreadyExists. (That approach also removes a race)